psi4/driver/qcdb/molecule.py

```diff
@@ -1336,6 +1336,87 @@ def run_dftd3(self, func: str = None, dashlvl: str = None, dashparam: Dict = Non
         elif derint == 1:
             return jobrec['extras']['qcvars']['DISPERSION CORRECTION GRADIENT']
 
+    def run_gcp(self, func=None, dertype=None, verbose=False):
+        """Compute geometrical BSSE correction via Grimme's GCP program.
+
+        Function to call Grimme's GCP program
+        https://www.chemie.uni-bonn.de/pctc/mulliken-center/software/gcp/gcp
+        to compute an a posteriori geometrical BSSE correction to *self* for
+        several HF, generic DFT, and specific HF-3c and PBEh-3c method/basis
+        combinations, *func*. Returns energy if *dertype* is 0, gradient
+        if *dertype* is 1, else tuple of energy and gradient if *dertype*
+        unspecified. The gcp executable must be independently compiled and
+        found in :envvar:`PATH` or :envvar:`PSIPATH`. *self* may be either a
+        qcdb.Molecule (sensibly) or a psi4.Molecule (works b/c psi4.Molecule
+        has been extended by this method py-side and only public interface
+        fns used) or a string that can be instantiated into a qcdb.Molecule.
+
+        Parameters
+        ----------
+        func : str, optional
+            Name of functional (func only, func & disp, or disp only) for
+            which to compute correction (e.g., blyp, BLYP-D2, blyp-d3bj,
+            blyp-d3(bj), hf+d).
+        dertype : int or str, optional
+            Maximum derivative level at which to run GCP. For large
+            molecules, energy-only calculations can be significantly more
+            efficient. Influences return values, see below.
+        verbose : int, optional
+            Amount of printing.
+
+        Returns
+        -------
+        energy : float
+            When `dertype=0`, energy [Eh].
+        gradient : ndarray
+            When `dertype=1`, (nat, 3) gradient [Eh/a0].
+        (energy, gradient) : tuple of float and ndarray
+            When `dertype=None`, both energy [Eh] and (nat, 3) gradient [Eh/a0].
+
+        """
+        import qcengine as qcng
+
+        if dertype is None:
+            derint, derdriver = -1, 'gradient'
+        else:
+            derint, derdriver = parse_dertype(dertype, max_derivative=1)
+
+        resinp = {
+            'molecule': self.to_schema(dtype=2),
+            'driver': derdriver,
+            'model': {
+                'method': func,
+                'basis': '(auto)',
+            },
+            'keywords': {
+                'verbose': verbose,
+            },
+        }
+        jobrec = qcng.compute(resinp, 'gcp', raise_error=True)
+        jobrec = jobrec.dict()
+
+        # hack as not checking type GRAD
+        for k, qca in jobrec['extras']['qcvars'].items():
+            if isinstance(qca, (list, np.ndarray)):
+                jobrec['extras']['qcvars'][k] = np.array(qca).reshape(-1, 3)
+
+        if isinstance(self, Molecule):
+            pass
+        else:
+            from psi4 import core
+
+            for k, qca in jobrec['extras']['qcvars'].items():
```

Comment:
Can we rename `k` and `qca` to `variable_name` and `value`, or something similar? This should happen above as well. It's more readable for those of us who aren't in-tune with QCEngine conventions.